koschei/backend/services/resolver.py

```diff
@@ -610,31 +623,67 @@ def process_builds(self, collection):
             .update({'deps_resolved': False}, synchronize_session=False)
         self.db.commit()
 
-    def process_repos(self, collection):
-        curr_repo = koji_util.get_latest_repo(
+    def get_new_repo_id(self, collection):
+        """
+        Returns a latest repo id that is suitable for new repo resolution or None.
+
+        :param: collection for which collection to query
+        """
+        latest_repo = koji_util.get_latest_repo(
             self.session.secondary_koji_for(collection),
-            collection.build_tag)
-        if curr_repo and collection.secondary_mode:
-            backend.refresh_repo_mappings(self.session)
-            mapping = self.db.query(RepoMapping)\
-                .filter_by(secondary_id=curr_repo['id'])\
-                .first()
-            # don't resolve it if we don't have it on primary yet
-            if not (mapping and mapping.primary_id and
-                    self.session.koji('primary')
-                    .getTaskInfo(mapping.task_id)['state'] ==
-                    koji.TASK_STATES['CLOSED']):
-                return
-        if curr_repo and curr_repo['id'] > collection.latest_repo_id:
-            repo_id = curr_repo['id']
-            try:
-                with self.prepared_repo(collection, repo_id) as sack:
-                    self.generate_repo(sack, collection, repo_id)
-            except RepoGenerationException as e:
-                self.log.exception("Cannot generate new repo (repo_id={})"
-                                   .format(repo_id), e)
+            collection.build_tag,
+        )
+
+        if latest_repo and latest_repo.get('id', 0) > collection.latest_repo_id:
+            if not collection.secondary_mode:
+                return latest_repo['id']
+            else:
+                # in secondary mode, we want to only resolve the repo if it was
+                # already regenerated on primary
+                backend.refresh_repo_mappings(self.session)
+                mapping = self.db.query(RepoMapping)\
+                    .filter_by(secondary_id=latest_repo['id'])\
+                    .first()
+                if (
+                        mapping and
+                        mapping.primary_id and
+                        (
+                            self.session.koji('primary')
+                            .getTaskInfo(mapping.task_id)['state']
+                        ) == koji.TASK_STATES['CLOSED']
+                ):
```

Comment:
For parenthesised expressions that don't fit on one line, I previously used
```
self.some_big_function(long, list,
                       of_complex_arguments)
```
Nowadays, I prefer to have one item per line
```
self.some_big_function(
    long,
    list,
    of_complex_arguments,
)
```

Don't you like this style?